### DISCUSSION TITLE
How to perform addition/sum on existing value with single update

### DISCUSSION THREAD

**Discussion by codestergit** [supabase/supabase#7613] [Questions] [answered]
# Bug report
I want to perform this
UPDATE table SET credit = credit + 7 WHERE id = 1

## Describe the bug
I found no documentation

## To Reproduce
Don't find any docs

Could you guys please help thanks

**Comment by Hallidayo** [comment, marked as answer]
Hi @codestergit

You could create a postgres function (rpc) to do this:

```
CREATE OR REPLACE FUNCTION count_function(rowid int, countvalue int) RETURNS void AS $$
  UPDATE "table" 
    SET credit = credit + countvalue 
    WHERE id = rowid;
$$ LANGUAGE sql;
```

Then you can execute the function on the frontend using JavaScript like so:

```
const { data, error } = await supabase
.rpc('count_function', { rowid: 1, countvalue: 7 })
```

Their are also the following guides/documentation on the Supabase website:

- [Database functions guide](https://supabase.com/docs/guides/database/functions)
- [Supabase Postgres function reference](https://supabase.com/docs/reference/javascript/rpc)

**Reply by steve-chavez** [reply]
Just FYI, [json patch](https://github.com/PostgREST/postgrest/issues/465#issuecomment-945971578) is being worked on and will allow doing increments purely on the client side.
